Add klicky probe support

Farva42 opened the pull request:
Add klicky probe support including stow/deploy menu.

### Description

Add new probe type from Klicky probes like the eluced, quickdraw, klackender, or other probes that require movements to deploy and stow the probe. Allows for probe to be used to home Z.

### Requirements

A probe that is deployed and stowed via movement.

### Benefits

Allows more probes to be used, and allows probes attached with movement to be used to home Z if they are able to.


### Related Issues
#24003
#24370

Farva42 committed:
Add klicky probe

Add klicky probe support including stow/deploy menu.

Farva42 committed:
Merge branch 'MarlinFirmware:bugfix-2.1.x' into bugfix-2.1.x

thisiskeithb commented on Marlin/Configuration.h:
```diff
@@ -1243,6 +1282,9 @@
 //
 
+//Add probe deploy and stow options to the motion menu.
+//#define PROBE_DEPLOY_STOW_MENU
```
When `BLTOUCH` is enabled, the menu items for deploy & stow are automatically enabled. I'd suggest this menu should also be enabled automatically for these new probes.

Farva42 replied:
I updated to make the default behaviour to enable the menu items with the option to disable.

DerAndere1 commented:
With a Little Bit of tweaking this Could also make the SAFE_BEDLEVELING_START_X ... Settings obsolete.

Farva42 committed:
Klicky Probe support tweaks

Change menu to enabled by default, removed un-needed sanity check.
Clean up menu.

Farva42 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/Farva42/Marlin_add_klicky into bugfix-2.1.x

Farva42 commented:
Well that's not going to work.

Farva42 committed:
Update menu_motion.cpp

revert menu changes

Farva42 committed:
Merge branch 'MarlinFirmware:bugfix-2.1.x' into bugfix-2.1.x

Farva42 committed:
Merge branch 'MarlinFirmware:bugfix-2.1.x' into bugfix-2.1.x

thinkyhead committed:
clean up

thinkyhead commented:
Always create a new branch for PRs so you don't pollute your default branch. Also, your bugfix-2.1.x branch is protected so PRs submitted from it cannot be fixed up.

Please merge https://github.com/Farva42/Marlin/pull/3 to update this PR.

Farva42 commented:
My apologies. This is my first time using GitHub and doing a PR definitely learned a few things for next time.

nionio6915 commented:
@thinkyhead what can Euclid team do to assist? We've had Euclids running under Marling for over a year despite the mixed stow/deploy behavior for G29, G35 and G28 gcodes.

Farva42 commented:
This accepts m401 and m402 commands. It also puts a deploy and Stow option in the motion menus(these use m401& m402)

I am going to start looking at how to clean up the behaviour with g28 followed by g29 as it will do a stow and deploy in between. It might take me a while as I'm not the experienced with coding. I've got an idea if what might be causing the inconsistency in positioning after Stow as well.

nionio6915 commented:
Look at issue #22794
https://github.com/MarlinFirmware/Marlin/issues/22794

We reported it and have tried to trace the source of all the differences.

nionio6915 commented:
How does this play with the G30 bug negating the E parameter?
